plugins/inputs/t128_peer_path/t128_peer_path.go

```diff
@@ -0,0 +1,120 @@
+package t128_peer_path
+
+import (
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/t128_graphql"
+)
+
+const (
+	//DefaultRequestTimeout is the request timeout if none is configured
+	DefaultRequestTimeout = 5 * time.Second
+
+	//timeoutDeadlineDiff is the difference between plugin.Timeout and the request deadline header
+	timeoutDeadlineDiff = 1 * time.Second
+)
+
+// T128PeerPath is an input for metrics of a 128T router instance
+type T128PeerPath struct {
+	CollectorName    string            `toml:"collector_name"`
+	BaseURL          string            `toml:"base_url"`
+	UnixSocket       string            `toml:"unix_socket"`
+	EntryPoint       string            `toml:"entry_point"`
```

Comment:
The config for the actual query etc. should not come from the user. If the user could configure it, they would just use the GQL input. These are specific to peer path and the plugin should not make them configurable. The only thing the user should configure are version specific fields which I'd make a boolean flag that the monitoring agent will ultimately set based on the running 128T version.

Reply:
That makes sense!